src/Module/Api/Friendica/Group/Create.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Api\Friendica\Group;
+
+use Friendica\Database\DBA;
+use Friendica\Model\Group;
+use Friendica\Module\BaseApi;
+use Friendica\Network\HTTPException;
+
+/**
+ * API endpoint: /api/friendica/group_create
+ */
+class Create extends BaseApi
+{
+	protected function post(array $request = [])
+	{
+		BaseApi::checkAllowedScope(BaseApi::SCOPE_WRITE);
+		$uid = BaseApi::getCurrentUserID();
+
+		// params
+		$name  = $this->getRequestValue($request, 'name', '');
+		$json  = json_decode($request['json'], true);
+		$users = $json['user'];
+
+		// error if no name specified
+		if ($name == '') {
+			throw new HTTPException\BadRequestException('group name not specified');
+		}
+
+		// error message if specified group name already exists
+		if (DBA::exists('group', ['uid' => $uid, 'name' => $name, 'deleted' => false])) {
```

Comment:
please add `Database` as constructor parameter and replace `DBA::` dynamic calls with `$this->dba->`

Reply:
A different idea would be to have a `Group` factory method, called `Group::exists($uid, $name)` and move the logic into it (separation of persistence logic from business logic)